### PR TITLE
feat(ToolbarItem): use compose()

### PR DESCRIPTION
**Pull request by mnajdova**
This PR adds compose functionality to `ToolbarItem`. 

# BREAKING CHANGES

Restructured `className` exports for the `ToolbarItem`'s `wrapper`, which is now a component.

### Before
```tsx
import { toolbarItemSlotClassNames } from '@fluentui/react-northstar';

console.log(toolbarItemSlotClassNames.wrapper);
```

### After
```tsx
import { toolbarItemWrapperClassName } from '@fluentui/react-northstar';

console.log(toolbarItemWrapperClassName);
```

### CODE REVIEW

File: packages/fluentui/react-northstar/src/components/Toolbar/ToolbarRadioGroup.tsx [hunk 1]
```diff
@@ -126,7 +127,7 @@ const ToolbarRadioGroup: React.FC<WithAsProp<ToolbarRadioGroupProps>> &
         return ToolbarDivider.create(item);
       }
 
-      const toolbarItem = ToolbarItem.create(item, {
+      const toolbarItem = createShorthand(ToolbarItem, item, {
```
Comment (mnajdova) [inline packages/fluentui/react-northstar/src/components/Toolbar/ToolbarRadioGroup.tsx:130]: `ToolbarItem.create` was removed

File: packages/fluentui/react-northstar/src/components/Toolbar/ToolbarMenu.tsx [hunk 2]
```diff
@@ -186,6 +186,9 @@ ToolbarMenu.defaultProps = {
 };
 
 ToolbarMenu.create = createShorthandFactory({ Component: ToolbarMenu, mappedArrayProp: 'items' });
+ToolbarMenu.shorthandConfig = {
```
Comment (mnajdova) [inline packages/fluentui/react-northstar/src/components/Toolbar/ToolbarMenu.tsx:189]: We need this in the `ToolbarItem`

Comment (mnajdova) [reply packages/fluentui/react-northstar/src/components/Toolbar/ToolbarMenu.tsx:189]: Can we removed after compose is added in `ToolbarMenu`

File: packages/fluentui/react-northstar/src/components/Toolbar/ToolbarItemIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ */
+const ToolbarItemIcon = compose<'div', ToolbarItemIconProps, ToolbarItemIconStylesProps, BoxProps, {}>(Box, {
+  className: toolbarItemIconClassName,
+  displayName: 'ToolbarItemIcon',
+
+  overrideStyles: true,
+  shorthandConfig: {
+    mappedProp: 'content',
+  },
+}) as ComponentWithAs<'div', ToolbarItemIconProps>;
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Toolbar/ToolbarItemIcon.tsx:23]:
```suggestion
});
```

File: packages/fluentui/react-northstar/src/components/Toolbar/ToolbarItemWrapper.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+ */
+const ToolbarItemWrapper = compose<'div', ToolbarItemWrapperProps, ToolbarItemWrapperStylesProps, BoxProps, {}>(Box, {
+  className: toolbarItemWrapperClassName,
+  displayName: 'ToolbarItemWrapper',
+
+  overrideStyles: true,
+  shorthandConfig: {
+    mappedProp: 'content',
+  },
+}) as ComponentWithAs<'div', ToolbarItemWrapperProps>;
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Toolbar/ToolbarItemWrapper.tsx:23]:
```suggestion
});
```

File: packages/fluentui/react-northstar/src/components/Toolbar/Toolbar.tsx [hunk 5]
```diff
@@ -458,27 +460,23 @@ const Toolbar = compose<'div', ToolbarProps, ToolbarStylesProps, {}, {}>(
...
-              onMenuOpenChange: (e, { menuOpen }) => {
-                _.invoke(props, 'onOverflowOpenChange', e, { ...props, overflowOpen: menuOpen });
-              },
-            },
-          })}
-        </Ref>
-      );
-    };
+    const renderOverflowItem = overflowItem =>
+      createShorthand(composeOptions.slots.overflowItem, overflowItem, {
```
Comment (mnajdova) [inline packages/fluentui/react-northstar/src/components/Toolbar/Toolbar.tsx:464]: Oops we forgot `ToolbarItem` here..

**Pull request merged**